client/transactions/list/index.tsx

```diff
@@ -550,6 +550,18 @@ export const TransactionsList = (
 				window.confirm( confirmMessage )
 			) {
 				try {
+					wcpayTracks.recordEvent(
+						wcpayTracks.events.TRANSACTIONS_DOWNLOAD_CSV_CLICK,
+						{
+							location: props.depositId
+								? 'deposit_details'
+								: 'transactions',
+							download_type: downloadType,
+							exported_transactions: rows.length,
+							total_transactions: transactionsSummary.count,
+						}
+					);
```

Comment:
Since this call to `recordEvent` will occur for both `endpoint`/`browser` download types, it could be placed _before_ the if/else statement, to be more DRY. And it would occur regardless of whether the `window.confirm()` call was confirmed by the user (closer to the `_click` action).

I'm happy to go with your judgement here.

Reply:
Thanks, Good spotting on this. I've moved this tracking and marked this as resolved.

Reply:
Nice, now the event is fired very close to user action, almost first thing in the `onClick` handler 👌